dcpy/lifecycle/builds/build.py

```diff
@@ -10,6 +15,97 @@
 STAGE = "builds.build"
 
 
+def export_dataset_from_postgres(
+    table_name: str,
+    file_path: Path,
+    format: ExportFormat,
+    pg_client: postgres.PostgresClient,
+) -> None:
+    """Import a recipe to local data library folder and build engine."""
+    logger.info(f"Exporting table {table_name} to {file_path} in format {format}")
+    match format:
+        case ExportFormat.dat:
+            with open(file_path, "wb") as f:
+                psql_process = subprocess.Popen(
+                    [
+                        "psql",
+                        pg_client.engine_uri,
+                        "--command",
+                        f"\\COPY (SELECT * FROM {table_name}) TO STDOUT;",
+                    ],
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                )
+
+                sed_process = subprocess.Popen(
+                    ["sed", r"s/$/\r/"],
+                    stdin=psql_process.stdout,
+                    stdout=f,
+                    stderr=subprocess.PIPE,
+                )
+
+                if psql_process.stdout:
+                    psql_process.stdout.close()
+                _output, _errors = sed_process.communicate()
+        case _:
+            raise NotImplementedError(
+                f"Export of dataset format {format} not implemented yet"
+            )
+
+
+def export(recipe_lock_path: Path) -> None:
+    recipe = plan.recipe_from_yaml(Path(recipe_lock_path))
+
+    if not recipe.exports:
+        logger.info("No exports defined in recipe, skipping export step")
+        return
+
+    build_name = metadata.build_name()
+    logger.info(f"Exporting build outputs for {recipe.name} build named {build_name}")
+
+    pg_client = postgres.PostgresClient(schema=build_name)
+    output_folder = Path(recipe.exports.output_folder)
```

Comment:
I think we should use the `dcpy.lifecycle.config.local_data_path_for_stage(...)` for the default base path, and pass a potential override as an argument to the function. So probably something like `.lifecycle/builds/build/export/product/{{ version + build identifier }}/`. Then might be nice to specify a subpath in the recipe?